Spring Boot Data JPA 1.0

aksoli666 opened the pull request:
used JpaRepository in BookRepository
added liquibase support
changed ddl-auto property value to validate
implemented soft delete concept for the Book
added endpoints for update book and for delete book

aksoli666 committed:
Spring Boot Data JPA 1.0
used JpaRepository in BookRepository
added liquibase support
changed ddl-auto property value to validate
implemented soft delete concept for the Book
added endpoints for update book and for delete book

Elena-Bruyako commented on src/main/java/shop/entity/Book.java:
```diff
@@ -30,7 +34,13 @@ public class Book {
     @Column(nullable = false)
     private BigDecimal price;
     private String description;
+    @Column(name = "cover_image")
```
```suggestion
```

Elena-Bruyako commented on src/main/java/shop/entity/Book.java:
```diff
@@ -30,7 +34,13 @@ public class Book {
+    @Column(
+            name = "is_deleted",
+            nullable = false
+    )
```
```suggestion
    @Column(nullable = false)
```

Elena-Bruyako commented on src/main/java/shop/repository/BookRepository.java:
```diff
@@ -1,13 +1,30 @@
+                       String isbn,
+                       BigDecimal price,
+                       String description,
+                       String coverImage);
```
```suggestion
```

Elena-Bruyako commented on src/main/java/shop/repository/BookRepository.java:
```diff
@@ -1,13 +1,30 @@
-    Optional<Book> findBookById(Long id);
-
-    List<Book> findAllBooks();
+@Repository
```
```suggestion
```

Elena-Bruyako commented on src/main/java/shop/service/impl/BookServiceImpl.java:
```diff
@@ -20,23 +21,41 @@ public class BookServiceImpl implements BookService {
     public BookDto getBookById(Long id) {
-        Book bookFromRepository = bookRepository.findBookById(id).orElseThrow(
-                        () -> new EntityNotFoundException("Can`t get book by id: " + id));
+        Book bookFromRepository = bookRepository.findById(id)
```
```suggestion
        Book book = bookRepository.findById(id)
```

Elena-Bruyako commented on src/main/java/shop/service/impl/BookServiceImpl.java:
```diff
@@ -20,23 +21,41 @@ public class BookServiceImpl implements BookService {
+
+    @Override
+    public BookDto updateBookById(Long id, UpdateBookDto updateBookDto) {
+        if (bookRepository.updateBookById(id,
```
```suggestion
        Book book = bookRepository.findById(id).orElseThrow
        bookMapper.updateBookFromDto(bookDto, book);
        return bookMapper.toDto(bookRepository.save(book));
```

Elena-Bruyako commented on src/main/java/shop/service/impl/BookServiceImpl.java:
```diff
@@ -20,23 +21,41 @@ public class BookServiceImpl implements BookService {
+    }
+
+    @Override
+    public String deleteBookById(Long id) {
```
```suggestion
    public void deleteBookById(Long id) {
```

Elena-Bruyako commented on src/main/java/shop/controller/BookController.java:
```diff
@@ -33,4 +36,14 @@ public BookDto getBookById(@PathVariable Long id) {
+    }
+
+    @DeleteMapping("/{id}")
+    public String deleteBookById(@PathVariable Long id) {
```
```suggestion
    public void deleteBookById(@PathVariable Long id) {
```

Elena-Bruyako commented on src/main/java/shop/service/impl/BookServiceImpl.java:
```diff
@@ -20,23 +21,41 @@ public class BookServiceImpl implements BookService {
+    @Override
+    public String deleteBookById(Long id) {
+        bookRepository.deleteById(id);
+        return "Book deleted";
```
```suggestion
```

Elena-Bruyako reviewed (changes requested):
see comments

aksoli666 committed:
Spring Boot Data JPA 2.0
changed method update for book, was used BookMapper
changed methods update and getById
didnt use annotation with method name in book class

Elena-Bruyako commented on src/main/java/shop/entity/Book.java:
```diff
@@ -31,6 +35,8 @@ public class Book {
+    private boolean isDeleted = false;
 
     @Override
     public String toString() {
```
remove toString method

Elena-Bruyako commented on src/main/java/shop/mapper/BookMapper.java:
```diff
@@ -11,4 +14,8 @@ public interface BookMapper {
+
+    List<BookDto> toBookDtoList(List<Book> books);
+
+    Book updateBookFromDto(UpdateBookDto updateBookDto, @MappingTarget Book book);
```
```suggestion
    void updateBookFromDto(UpdateBookDto updateBookDto, @MappingTarget Book book);
```

Elena-Bruyako commented on src/main/java/shop/dto/request/UpdateBookDto.java:
```diff
@@ -0,0 +1,16 @@
+
+@Getter
+@Setter
+public class UpdateBookDto {
```
```suggestion
public class UpdateBookRequestDto {
```

Elena-Bruyako commented on src/main/java/shop/service/impl/BookServiceImpl.java:
```diff
@@ -20,23 +21,33 @@ public class BookServiceImpl implements BookService {
-        Book bookFromRepository = bookRepository.findBookById(id).orElseThrow(
-                        () -> new EntityNotFoundException("Can`t get book by id: " + id));
-        return bookMapper.toBookDto(bookFromRepository);
+        return bookMapper.toBookDto(bookRepository.findById(id).get());
```
```suggestion
        Book book = bookRepository.findBookById(id).orElseThrow(
                        () -> new EntityNotFoundException("Can`t get book by id: " + id));
        return bookMapper.toBookDto(book);
```

Elena-Bruyako commented on src/main/java/shop/service/impl/BookServiceImpl.java:
```diff
@@ -20,23 +21,33 @@ public class BookServiceImpl implements BookService {
+    @Override
+    public BookDto updateBookById(Long id, UpdateBookDto updateBookDto) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Book not found. Id: " + id));
+        return bookMapper.toBookDto(
+                bookRepository.save(
+                        bookMapper.updateBookFromDto(updateBookDto, book)
+                )
```
Let's improve readability
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("Book not found. Id: " + id));
        bookMapper.updateBookFromDto(updateBookDto, book)
        return bookMapper.toBookDto(bookRepository.save(book));
```

Elena-Bruyako commented on src/main/java/shop/service/impl/BookServiceImpl.java:
```diff
@@ -20,23 +21,33 @@ public class BookServiceImpl implements BookService {
     @Override
     public BookDto createBook(CreateBookDto createBookDto) {
```
```suggestion
    public BookDto createBook(CreateBookRequestDto createBookDto) {
```

Elena-Bruyako commented on src/main/resources/db/changelog/changes/01-create-books-table.yaml:
```diff
@@ -0,0 +1,46 @@
+                  name: is_deleted
+                  type: boolean
+                  constraints:
+                    nullable: false
```
add empty line at the end of file

Elena-Bruyako commented on src/main/resources/db/changelog/db.changelog-master.yaml:
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-books-table.yaml
```
add empty line at the end of file

Elena-Bruyako reviewed (changes requested):
see comments

aksoli666 committed:
Spring Boot Data JPA 3.0
removed toString() in Book.class
updateBookRequestDto() -> void
renamed request dto
modified methods -> findBookById, updateBookById
added empty line in changelog files
